recipe/meta.yaml

```
@@ -14,156 +15,228 @@
-    - cuda-version >={{ cuda_major }}.2,<{{ cuda_major+1 }}  # [(cuda_compiler_version or "").startswith("11")]
-    - cuda-version >={{ cuda_major }}.0,<{{ cuda_major+1 }}  # [(cuda_compiler_version or "").startswith("12")]
-
-  run_constrained:
-    # Only GLIBC_2.17 or older symbols present
-    - __glibc >=2.17  # [linux]
-    - scipy >=1.7,<2
-    - optuna >=3,<4
-    - __cuda >=11.2  # [(cuda_compiler_version or "").startswith("11")]
-    - __cuda >=12.0  # [(cuda_compiler_version or "").startswith("12")]
-    # TODO: Waiting for `aarch64` & `ppc64le` CUDA 12 support
-    - {{ pin_compatible('cudnn') }}                        # [not ((aarch64 or ppc64le) and (cuda_compiler_version or "").startswith("12"))]
-    - {{ pin_compatible('nccl') }}                         # [linux]
-    - {{ pin_compatible('cutensor', lower_bound='1.4') }}
-    - {{ pin_compatible('cusparselt', max_pin='x.x') }}    # [(linux64 or aarch64 or win) and (cuda_compiler_version or "").startswith("11")]
-
-test:
-  requires:
-    # need the libcuda stub for import test
-    - cuda-driver-dev  # [linux and (cuda_compiler_version or "").startswith("12")]
-
-  source_files:
-    - tests
+  - name: cupy-core
+    version: {{ version }}
+    build:
+      number: {{ build_num }}
+      skip: true  # [(py < 39) or cuda_compiler_version not in ("11.8", "12.0")]
+      script:
+        # CuPy default detects CUDA from nvcc, but on Conda-Forge's dockers nvcc lives in a different place...
+        # With conda-forge/nvcc-feedstock#58, CUDA_PATH is set correctly
+        - export NVCC="$(which nvcc)"                                                # [linux]
+        {% if cuda_major >= 12 %}
+        - export CUPY_NUM_BUILD_JOBS=1                                               # [(aarch64 or ppc64le) and (build_platform == target_platform)]
+        - export CUDA_PATH=$PREFIX/targets/{{ target_name }}                         # [linux]
+        {% endif %}
+        - echo "nvcc is $NVCC, CUDA path is $CUDA_PATH"                              # [linux]
+        - for /f "tokens=* usebackq" %%f in (`where nvcc`) do (set "dummy=%%f" && call set "NVCC=%%dummy:\=\\%%")  # [win]
+        # we use this WAR until we figure out how to proceed with conda-forge/nvcc-feedstock#88
+        - set "NVCC=%NVCC% --use-local-env"  # [win64]
+        - echo "nvcc is %NVCC%, CUDA path is %CUDA_PATH%"  # [win]
+        - export CUSPARSELT_PATH="${PREFIX}"                                         # [(linux64 or aarch64 or win) and (cuda_compiler_version or "").startswith("11")]
+        # Workaround __ieee128 error; see https://github.com/LLNL/blt/issues/341
+        - export NVCC="$NVCC -Xcompiler -mno-float128"  # [ppc64le]
+  
+        # {{ PYTHON }} is not resolved properly in multi-output recipes...
+        - $PREFIX/bin/python -m pip install . --no-deps -vv                          # [linux]
+        - if 1 == 1 %PREFIX%\bin\python -m pip install . --no-deps -vv               # [win]
```

@jakirkham any suggestion to fix the syntax for Windows? Yaml parser wouldn't allow beginning with `%`, so
``` 
- %PREFIX%\bin\python -m pip install . --no-deps -vv               # [win]
```
doesn't work

---

Does using the environment variable work (recognize the Jinja variable has issues, but the environment variable may behave better)?

```suggestion
        - %PYTHON% -m pip install . --no-deps -vv                                    # [win]
```

---

No, that would be the same as starting with `%PREFIX%\bin\python ...`. Yaml disallows this, thus me asking a workaround 😞

---

To be more clear, using environment variables are fine, we use it elsewhere in the recipe, but they cannot be at the beginning of the statement, Yaml has a hard time to understand it.

---

https://github.com/conda-forge/cupy-feedstock/pull/241#issuecomment-1900980100

---

I see. Could we set the `PYTHON` Jinja value ourselves and then use that here?

---

So the fix is in commit [8d9012f](https://github.com/conda-forge/cupy-feedstock/pull/241/commits/8d9012fa83a796940652efa8a214f3a3598f89cc). Turns out on Windows the Python executable is not in `$PREFIX/bin`...

---

Another option is to do something [like this]( https://github.com/conda-forge/r-websocket-feedstock/blob/d01855f106975df76fbbc47e7e4856c449d2c537/recipe/meta.yaml#L58 ), but it is a bit clunky

---

Ok pushed a couple commits, which should help address these issues

---

Looks good, thanks John!